states/states/graph.c

```diff
@@ -0,0 +1,136 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "graph.h"
+
+typedef struct Graph
+{
+    List** capitals;
+    int numberOfCapitals;
+    int numberOfCities;
+    int matrix[NUMBER_OF_CITIES][NUMBER_OF_CITIES];
+} Graph;
+
+Graph* createGraph(void)
+{
+    Graph* graph = calloc(1, sizeof(Graph));
+    graph->capitals = calloc(NUMBER_OF_CITIES, sizeof(List*));
+    if (graph->capitals == NULL)
+    {
+        free(graph);
+        return NULL;
+    }
+    graph->numberOfCities = 0;
+    return graph;
+}
+
+void deleteGraph(Graph** graph)
+{
+    int numberOfCapitals = (*graph)->numberOfCapitals;
+    for (int i = 0; i < numberOfCapitals; i++)
+    {
+        deleteList(&((*graph)->capitals[i]));
+    }
+    free(*graph);
+    *graph = NULL;
+}
+
+int getEdge(Graph* graph, int start, int end)
+{
+    return graph->matrix[start][end];
+}
+
+void setEdge(Graph* graph, int firstNode, int secondNode, int weight)
+{
+    graph->matrix[firstNode][secondNode] = weight;
+    graph->matrix[secondNode][firstNode] = weight;
+}
+
+Graph* readFile(char* fileName)
+{
+    Graph* graph = createGraph();
+    if (graph == NULL)
+    {
+        return NULL;
+    }
+    FILE* input = fopen(fileName, "r");
+    int numberOfCities = 0;
+    int numberOfRoads = 0;
+    fscanf_s(input, "%d%*c", &numberOfCities);
```

Comment:
```suggestion
    fscanf_s(input, "%d", &numberOfCities);
```
При чтении с форматным спецификатором %d пробельные символы и даже переводы строк сами пропустятся. Проблемы-то могут быть только со считыванием строк чем-нибудь хитрым типа `[^\n]`